### ISSUE TITLE
When the country prop changes programmatically, the <PhoneInput> and <CountrySelect> components do not update accordingly to reflect this change

### ISSUE THREAD

**Issue by emmanuel-isaac** [catamphetamine/react-phone-number-input#224]
If at any time during the PhoneInput component's lifecycle, the country prop changes, the flag remains the same, and the input mask does not get updated.

**Comment by catamphetamine** [comment]
If any value has been entered, then no.

**Comment by catamphetamine** [comment]
You can only set a country when the input is empty.

**Comment by emmanuel-isaac** [comment]
There's a use-case where after a phone has been entered, its value could indicate a different country other than the currently selected country. This is particularly possible for American and Canadian phone numbers that have similar formats but only differ in the area code.

**Comment by emmanuel-isaac** [comment]
In that case, it'd be helpful to programmatically switch the flag to the appropriate country, don't you think?

**Comment by catamphetamine** [comment]
Provide an exact example which can be reproduced on the demo page

**Comment by emmanuel-isaac** [comment]
Select "United States" from the country dropdown, then enter "6045555555" (which is a Canadian number). There's no way to switch the flag to a Canadian flag.

**Comment by catamphetamine** [comment]
I see.
Ok, I'll look into this this/next week maybe.

**Comment by emmanuel-isaac** [comment]
I think it's a simple fix unless you think more needs to be done. I can create a PR

**Comment by catamphetamine** [comment]
@emmanuel-isaac 
> I think it's a simple fix unless you think more needs to be done. I can create a PR

I know another simple fix: `onChange()` handler will parse `value` and determine its `country`.
If `value`'s `country` is not `undefined` then it will switch the country automatically.

**Comment by catamphetamine** [comment]
Currently it does so if you enter "+1 604 555 5555".
The same behavior will be replicated for non-international numbers, I guess.

**Comment by emmanuel-isaac** [comment]
Yeah @catamphetamine . This should work. I have something locally. Could you highlight some steps on how to run a demo locally?

**Comment by catamphetamine** [comment]
`npm run build`, then open `bundle/index.html`.

**Comment by emmanuel-isaac** [comment]
```
➜  react-phone-number-input git:(master) ✗ find . -type f -not -path "*node_modules/*" -name "index.html"
➜  react-phone-number-input git:(master) ✗
```

No results

**Comment by catamphetamine** [comment]
oh, seems that it's not in the git.
yeah, i guess so.
then switch to `gh-pages` and copy all files and `lib` folder from there to "build" folder of the master branch.

![image](https://user-images.githubusercontent.com/477167/50653662-52805380-0f9b-11e9-8aed-5c3de08f9035.png)

**Comment by emmanuel-isaac** [comment]
Could you give it a try?

**Comment by emmanuel-isaac** [comment]
Okay @catamphetamine . Thanks. I'll try it out

**Comment by emmanuel-isaac** [comment]
LGTM!
